contracts/LpSugar.vy

```diff
@@ -1132,12 +1152,35 @@ def _cl_lp(_data: address[4], _token0: address, _token1: address) -> Lp:
   token1_fees: uint256 = 0
   locked: uint256 = staticcall locker_factory.locked(_data[1])
   emerging: bool = staticcall self.cl_launcher.emerging(_data[1])
+  created_at: uint32 = 0
 
   slot: Slot = staticcall pool.slot0()
   tick_low: int24 = (slot.tick // tick_spacing) * tick_spacing
   tick_high: int24 = tick_low + tick_spacing
 
-  if gauge_liquidity > 0 and gauge.address != empty(address):
+  if gauge.address == empty(address):
+    launcher_pool: PoolLauncherPool = staticcall self.cl_launcher.pools(_data[1])
+
+    if launcher_pool.created_at != 0:
+      created_at = launcher_pool.created_at
+      
+      # fetch new and old observations from pool oracle
+      obs_new: Observation = staticcall pool.observations(slot.observationIndex)
+      obs_old: Observation = staticcall pool.observations(0)
+
+      if slot.cardinality >= slot.cardinalityNext:
+        old_index: uint16 = (slot.observationIndex + 1) % slot.cardinality
+        obs_old = staticcall pool.observations(old_index)
+
+      # compute time delta and seconds per liquidity delta
+      time_delta: uint256 = convert(obs_new.blockTimestamp, uint256) - convert(obs_old.blockTimestamp, uint256)
+      splc_delta: uint256 = convert(obs_new.secondsPerLiquidityCumulativeX128, uint256) - convert(obs_old.secondsPerLiquidityCumulativeX128, uint256)
+
+      avg_liquidity: uint256 = shift(time_delta, 128) / splc_delta
+
+      token0_fees = (staticcall pool.feeGrowthGlobal0X128() / 2**128) * avg_liquidity
+      token1_fees = (staticcall pool.feeGrowthGlobal1X128() / 2**128) * avg_liquidity
```

Comment:
I think we'd have to bring these per second first or return the time delta in order to know the "denominator" :thinking: